core/rust/utils/src/assertions.rs

```diff
@@ -47,6 +62,14 @@ pub fn assert_derivation(
 ) -> Result<u8, ProgramError> {
     let (key, bump) = Pubkey::find_program_address(path, program_id);
     if key != *account.key {
+        msg!(
+            "derivation assertion failed for {actual_key}:\n
```

Comment:
"invalid derivation {actual_key} != {key}"

Reply:
Looks like `\n` is not needed, I'll remove that.